src/open_inwoner/openzaak/clients.py

```diff
@@ -692,43 +702,205 @@ def fetch_open_tasks(self, bsn: str) -> list[OpenTask]:
         return results
 
 
+TClient = TypeVar("TClient", bound=APIClient)
+
+
+@dataclass(frozen=True)
+class ZgwClientResponse:
+    """A single response in a MultiZgwClientResult."""
+
+    client: TClient
+    result: Any
+    exception: Exception | None = None
+
+
+@dataclass(frozen=True)
+class MultiZgwClientProxyResult:
+    """Container for a multi-backend responses"""
+
+    responses: list[ZgwClientResponse]
+
+    @cached_property
+    def has_errors(self) -> bool:
+        return any(r.exception is not None for r in self.responses)
+
+    @cached_property
+    def failing_responses(self) -> list[ZgwClientResponse]:
+        return list(r for r in self if r.exception is not None)
+
+    @cached_property
```

Comment:
I'm not sure this is workable in the context of uwsgi, where we have threading active within the same process. In-memory caching without taking this into account is a bit of a minefield, please err on the side of caution considering our domain